oak_functions/sdk/oak_functions/src/lib.rs

```diff
@@ -0,0 +1,141 @@
+        };
+
+        let status = OakStatus::from_i32(status_code as i32);
+        match status {
+            Some(status) => match status {
+                OakStatus::Ok => {
+                    unsafe {
+                        // The read operation succeeded, and overwrote some fraction of the vector's
+                        // available capacity with returned data (possibly zero). As the data is
+                        // already present in the vector, set its length to match what's available.
+                        buf.set_len(actual_size as usize);
+                    }
+                    return Ok(());
+                }
+                OakStatus::ErrBufferTooSmall if !(*resized) => {
+                    // Extend the vector to be large enough for the message
+                    debug!(
+                        "Got space for {} bytes, need {}",
+                        buf.capacity(),
+                        actual_size
+                    );
+                    if (actual_size as usize) > buf.capacity() {
+                        let extra = (actual_size as usize) - buf.len();
+                        buf.reserve(extra);
+                    }
+
+                    // Try again with a buffer resized to cope with expected size of data.
+                    continue;
+                }
+                status => {
+                    return Err(status);
+                }
+            },
+            None => return Err(OakStatus::ErrInternal),
+        }
+    }
+    error!("unreachable code reached");
+    Err(OakStatus::ErrInternal)
+}
+
+/// Write the response body.
+///
+/// Multiple calls to this function will replace the earlier responses. Only the last response that
+/// is written will be kept and returned to the user.
+pub fn write_response_body(response_body: &str) -> Result<(), OakStatus> {
+    write_response(response_body.as_bytes())
+}
+
+fn write_response(buf: &[u8]) -> Result<(), OakStatus> {
+    let status = unsafe { oak_functions_abi::write_response(buf.as_ptr(), buf.len()) };
+    result_from_status(status as i32, ())
+}
+
+/// Convert a status returned from a host function call to a `Result`.
+///
+/// The status is interpreted as an int representing an `OakStatus` enum value.
+///
+/// If the status is [`OakStatus::Ok`] then returns the provided value as [`Result::Ok`], otherwise
+/// returns the status as [`Result::Err`].
+///
+/// Note that host function calls usually return an `u32` because of limitations of the Wasm type
+/// system, so these values would usually be converted (via a cast) to `i32` by callers.
+pub fn result_from_status<T>(status: i32, val: T) -> Result<T, OakStatus> {
+    match OakStatus::from_i32(status) {
+        Some(OakStatus::Ok) => Ok(val),
+        Some(status) => Err(status),
+        None => Err(OakStatus::Unspecified),
+    }
+}
+
+/// Install a panic hook that logs [panic information].
```

Comment:
Where does it log it? AFAICT there is no logging at all from functions, or are you suggesting we should add it?